[RDY] cmake: don't override cmakeDir when set

teto opened the pull request:
When dontUseCmakeBuildDir is true (aka the default), it overrides
cmakeDir regardless of the package configuration.

While packaging netbee, I needed to both keep dontUseCmakeBuildDir to
true (some hardcoded paths expect the build folder) and set cmakeDir
(since CMakeList.txt was in a subfolder) which proved impossible.
Here is the fix.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40687&attempt_id=ed6a609c-30b3-46f6-8fc5-40b06884f9cd)

Attempted: cmake

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/bin/cpack
shrinking /nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/bin/ctest
shrinking /nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/bin/cmake
strip is /nix/store/jk6j4lh9v5mvjdbdc35sj0zffhhf6s56-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/bin
patching script interpreter paths in /nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2
/nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/share/cmake-3.10/Modules/CPack.STGZ_Header.sh.in: interpreter directive changed from "/bin/sh" to "/nix/store/l4w7xwjy2nmk31fl5kgyy7gg8z7l9n8z-bash-4.4-p19/bin/sh"
/nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/share/cmake-3.10/Modules/Squish4RunTestCase.sh: interpreter directive changed from "/bin/sh" to "/nix/store/l4w7xwjy2nmk31fl5kgyy7gg8z7l9n8z-bash-4.4-p19/bin/sh"
/nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2/share/cmake-3.10/Modules/SquishRunTestCase.sh: interpreter directive changed from "/bin/sh" to "/nix/store/l4w7xwjy2nmk31fl5kgyy7gg8z7l9n8z-bash-4.4-p19/bin/sh"
checking for references to /build in /nix/store/5swiv9yffrx052zmdfis55kwglyb64sl-cmake-3.10.2...
```
</p></details>

LnL7 reviewed (approved):
Looks good to me, but this should go to staging first.

teto committed:
cmake: don't override cmakeDir when set

When dontUseCmakeBuildDir is true (aka the default), it overrides
cmakeDir regardless of the package configuration.

While packaging netbee, I needed to both keep dontUseCmakeBuildDir to
true (some hardcoded paths expect the build folder) and set cmakeDir
(since CMakeList.txt was in a subfolder) which proved impossible.
Here is the fix.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40687&attempt_id=f4395ed9-901a-4162-abfa-75df9f6865d7)

Attempted: cmake

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/bin/cmake
shrinking /nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/bin/ctest
shrinking /nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/bin/cpack
strip is /nix/store/mgyvr3kil9qv0mywikpi0dbli061bpg4-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/bin
patching script interpreter paths in /nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2
/nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/share/cmake-3.10/Modules/SquishRunTestCase.sh: interpreter directive changed from "/bin/sh" to "/nix/store/3r4wp5drzynkz6hccmjpifkzwgqbs5nb-bash-4.4-p19/bin/sh"
/nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/share/cmake-3.10/Modules/CPack.STGZ_Header.sh.in: interpreter directive changed from "/bin/sh" to "/nix/store/3r4wp5drzynkz6hccmjpifkzwgqbs5nb-bash-4.4-p19/bin/sh"
/nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2/share/cmake-3.10/Modules/Squish4RunTestCase.sh: interpreter directive changed from "/bin/sh" to "/nix/store/3r4wp5drzynkz6hccmjpifkzwgqbs5nb-bash-4.4-p19/bin/sh"
checking for references to /build in /nix/store/0mi0j5iw3q5q18xrxsq1j0fwk99s06ak-cmake-3.10.2...
```
</p></details>

teto commented:
I hadn't realize I could change the branch myself. It's done.

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40687&attempt_id=431a99e4-4a74-49e1-a2ff-7622c4b13a78)

Attempted: cmake

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking /nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/bin/cpack
shrinking /nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/bin/ctest
shrinking /nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/bin/cmake
strip is /nix/store/8yfik687kfccisxnad42j19lfb7ij9b4-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/bin
patching script interpreter paths in /nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2
/nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/share/cmake-3.10/Modules/CPack.STGZ_Header.sh.in: interpreter directive changed from "/bin/sh" to "/nix/store/vnb8q2h7951gd551nm2vq2g6n8296g5b-bash-4.4-p19/bin/sh"
/nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/share/cmake-3.10/Modules/Squish4RunTestCase.sh: interpreter directive changed from "/bin/sh" to "/nix/store/vnb8q2h7951gd551nm2vq2g6n8296g5b-bash-4.4-p19/bin/sh"
/nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2/share/cmake-3.10/Modules/SquishRunTestCase.sh: interpreter directive changed from "/bin/sh" to "/nix/store/vnb8q2h7951gd551nm2vq2g6n8296g5b-bash-4.4-p19/bin/sh"
checking for references to /build in /nix/store/ipwrqln1ln9d8gsb1mm75rymwwl3lqw6-cmake-3.10.2...
```
</p></details>

teto commented:
@LnL7 is that ok ?
